packages/odyssey-react-mui/src/web-component/shadow-dom.ts

```diff
@@ -10,28 +10,40 @@
  * See the License for the specific language governing permissions and limitations under the License.
  */
 
-export const createShadowDomElements = (containerElement: HTMLElement) => {
-  const shadowRoot = containerElement.attachShadow({ mode: "open" });
-
-  // Container for Emotion `<style>` elements.
+export const createUnattachedShadowDomElements = () => {
+  const appRootElement = document.createElement("div");
   const emotionRootElement = document.createElement("div");
+
+  // This `div` may cause layout issues unless it inherits the parent's height.
+  appRootElement.style.setProperty("height", "inherit");
+
+  appRootElement.setAttribute("id", "app-root");
   emotionRootElement.setAttribute("id", "style-root");
   emotionRootElement.setAttribute("nonce", window.cspNonce);
 
-  // React app root component.
-  const shadowRootElement = document.createElement("div");
-  shadowRootElement.setAttribute("id", "shadow-root");
-  // This div could cause issues with layout of children.
-  // For flexibility, make it inherit its parent's height
-  shadowRootElement.style.setProperty("height", "inherit");
+  return { appRootElement, emotionRootElement };
+};
+
+export type ShadowDomElements = ReturnType<
+  typeof createUnattachedShadowDomElements
+>;
+
+export const createShadowDomElements = (containerElement: HTMLElement) => {
```

Comment:
Given what you wrote below, do you want a @deprecated here?

FWIW I'm not convinced this should be deprecated but we should test as we were discussing offline.

Reply:
dc18d415e52ab9f89e310e7df7dded938ae53013